### PR TITLE
Feature/2444 check certificates

### LINKED ISSUE #2444: SSL certificates expiration monitoring
Due to the impact that an expired SSL certificate could have on a PacketFence setup, it would be great to have a monitoring / alerting job that take care of checking the validity of it.

### PR DESCRIPTION
**Pull request by dwlfrth**
# Description
Checks for SSL certificates expiration and alert

# Impacts
pfmon

# Code / PR Dependencies
starttls

# Issue
fixes #2444 

# Delete branch after merge
YES

# NEWS file entries
## Bug Fixes
* SSL certificates expiration monitoring (#2444)

### CODE REVIEW

**Comment by dwlfrth** [comment]
Not ready because it need the starttls integration for sending mails

**Comment by jrouzierinverse** [comment]
If you use pf::config::util::pfmailer function to send out your email there is no need to wait for starttls. 
Since they accept the same arguments.

**Comment by dwlfrth** [comment]
Ready for review

File: conf/pfmon.conf.defaults [hunk 1]
```diff
@@ -568,3 +568,40 @@ interval=10m
...
+#
+# httpd_certificate
+#
+# SSL certificate used by HTTPd instance(s)
+httpd_certificate=/usr/local/pf/conf/ssl/server.pem
+#
+# radiusd_certificate
+#
+# SSL certificate used by RADIUSd instance(s)
+radiusd_certificate=/usr/local/pf/raddb/certs/server.crt
```
Comment (jrouzierinverse) [inline conf/pfmon.conf.defaults]: To "future proof" the feature having a list of certificates will allow additional certificates to be added without change any code.

```
certificates=/usr/local/pf/conf/ssl/server.pem,/usr/local/pf/raddb/certs/server.crt
```

File: lib/pf/pfmon/task/certificates_check.pm [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        # Alert for expired certificates
+        if ( $problematic_certs{$cert}{'expired'} ) {
+            $message = "SSL certificate '$cert' is expired. This should be addressed to avoid issues.";
+        }
+        # Alert for certificates that expires soon
+        elsif ( $problematic_certs{$cert}{'expires_soon'} ) {
+            $message = "SSL certificate '$cert' is about to expire soon (less than '" . $self->delay . "'). This should be taken care.";
+        }
+        $logger->warn($message);
+        pf::config::util::pfmailer(('subject' => "SSL certificate expiration", 'message' => $message));
```
Comment (jrouzierinverse) [inline lib/pf/pfmon/task/certificates_check.pm:77]: Should we just send one email instead of multiple.

File: conf/pfmon.conf.defaults (hunk 1 above)
Comment (dwlfrth) [reply conf/pfmon.conf.defaults]: Lot better yes, will adjust

File: lib/pf/pfmon/task/certificates_check.pm (hunk 2 above)
Comment (dwlfrth) [reply lib/pf/pfmon/task/certificates_check.pm:77]: I'd say, one per cert to make sure they get alerted and they know they are alerted for each cert

**Comment by dwlfrth** [comment]
Re-review

File: html/pfappserver/lib/pfappserver/Form/Config/Pfmon/certificates_check.pm [hunk 3]
```diff
@@ -27,26 +27,17 @@ has_field 'delay' => (
     },
 );
 
-has_field 'httpd_certificate' => (
+has_field 'certificates' => (
     type            => 'Text',
```
Comment (rouzier) [inline html/pfappserver/lib/pfappserver/Form/Config/Pfmon/certificates_check.pm]: TextArea

**Comment by dwlfrth** [comment]
Re-review

**Pull request merged**